### PR TITLE
Clarify docs of TargetProvider

### PR DESCRIPTION
**Pull request by fabxc**
@brian-brazil @jimmidyson

### CODE REVIEW

File: retrieval/targetmanager.go [hunk 1]
```diff
@@ -33,7 +33,10 @@ import (
 //
 // The TargetProvider does not have to guarantee that an actual change happened.
 // It does guarantee that it sends the new TargetGroup whenever a change happens.
-// On startup it sends all TargetGroups it can see.
+//
+// Sources() is guaranteed to be called before each call to Run(). On a call to Run()
+// implementing types must send a valid target group for each of the sources they
```
Comment (jimmidyson) [inline retrieval/targetmanager.go]: What if don't? "Must" might be too strong. What happens if they send through a `TargetGroup` that is not related to a source?

Comment (fabxc) [inline retrieval/targetmanager.go]: They may send more, which this statement does not exclude. The source is always part of the `config.TargetGroup` so sending a TG not related to a source is basically not possible.

Comment (fabxc) [inline retrieval/targetmanager.go]: To clarify, the `Sources()` are just a way to determine which target groups from before a reload not to drop.

Comment (jimmidyson) [inline retrieval/targetmanager.go]: OK cool.

File: retrieval/targetmanager.go [hunk 2]
```diff
@@ -33,7 +33,10 @@ import (
 //
 // The TargetProvider does not have to guarantee that an actual change happened.
 // It does guarantee that it sends the new TargetGroup whenever a change happens.
-// On startup it sends all TargetGroups it can see.
+//
+// Sources() is guaranteed to be called before each call to Run(). On a call to Run()
```
Comment (brian-brazil) [inline retrieval/targetmanager.go]: called exactly once

**Comment by brian-brazil** [comment]
:+1:
